internal/raft/replicator_test.go

```diff
@@ -287,12 +287,12 @@ func (this *cluster) assertMembers(t *testing.T, members []string) {
 
 func (peer *peer) assertMembers(t *testing.T, leader string, members []string) {
 	leaderNode, clusNodes := peer.repl.ListMembers()
-	if clusNodes[leaderNode] != leader {
+	if clusNodes[leaderNode].NodeUrl != leader {
```

Comment:
Check if its possible to verify other states/node types. You can consider including these assertions into other tests where additional scenarios are tested.

Reply:
Added tests.